src/main/kotlin/account/Account.kt

```diff
@@ -5,44 +5,45 @@ class Account(
     val conta: Int,
     val saldoInicial: Double
 ) {
-    var saldo = 0.0
+    var saldo  = saldoInicial
         protected set
     //protected faz com o valor do saldo nao seja modificado em outros arquivos, mas permite que as classes "filhas"
     // a utilizem
 
     fun deposita(valor: Double){
 
-        if(valor <= 0.0){
-            this.saldo += valor + saldoInicial
-        }
-        else if(valor > 0 && valor <= 10_000){
+        if(valor > 0.0 && valor <= 10_000.0){
             this.saldo += valor
         } else{
             throw error("valor não permitido")
         }
     }
 
     fun saque(valor: Double){
-        if(saldo >= valor){
-            this.saldo -= valor;
-            if(saldo < 0){
+        if(valor <= saldo){
+            if(valor > 0.0){
```

Comment:
Daria para diminuir a duplicação fazendo um if só
```suggestion
        if(valor <= saldo && valor > 0.0){
```